### PR TITLE
2015 09 jkarni travis cache

### LINKED ISSUE #178: Switch to container-based travis
Opening this issue so we can gather data / discuss / work on this incrementally.  (Also so that everyone knows we know.)

### PR DESCRIPTION
**Pull request by jkarni**
Fixes #178.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,12 +33,18 @@ before_install:
     - make init
     - make start
     - cd ../..
+
 install:
-    - ./misc/thentos-install.sh
+    - travis_retry cabal update
+    # We install hlint because we don't want to invalidate it's cache together
```
Comment (fhartwig) [inline .travis.yml]: "its" ;-)

File: .travis.yml [hunk 2]
```diff
@@ -15,12 +33,18 @@ before_install:
...
+    # with the thentos cache
     - git clone --depth 1 https://github.com/liqd/travis-cache
     - export PATH=$PWD/travis-cache:$PATH
     - mkdir -p ./hlint/.cabal-sandbox/share/x86_64-linux-ghc-7.8.4
     - cp -r travis-cache/data ./hlint/.cabal-sandbox/share/x86_64-linux-ghc-7.8.4/hlint-1.9.21/
+    - sed -i 's/^jobs:/-- jobs:/' ${HOME}/.cabal/config
```
Comment (fhartwig) [inline .travis.yml]: Maybe add a comment to explain this.

**Comment by fhartwig** [comment]
Looks good. Do you want to squash the experimental commits?

**Comment by jkarni** [comment]
I'll squash everything.

**Pull request merged**